packages/react-router/lib/rsc/browser.tsx

```diff
@@ -457,7 +465,7 @@ function createRouteFromServerManifest(
   let hasInitialError = payload?.errors && match.id in payload.errors;
   let initialError = payload?.errors?.[match.id];
   let isHydrationRequest =
-    match.clientLoader?.hydrate === true || !match.hasLoader;
+    match.clientLoader?.hydrate === true || !match.hasLoader || !match.element;
```

Comment:
TODO: Add comment explaining this. We need to explore what API routes mean in an RSC world.